megalinter/MegaLinter.py

```diff
@@ -822,7 +822,7 @@ def display_header():
         logging.info("")
 
     def check_results(self):
-        print(f"::set-output name=has_updated_sources::{str(self.has_updated_sources)}")
+        print(f'"has_updated_sources={str(self.has_updated_sources)}" >>"$GITHUB_OUTPUT"')
```

Comment:
```suggestion
        add_to_output_cmd = f'echo "has_updated_sources={str(has_updated_sources)}" >>"$GITHUB_OUTPUT"'
        add_to_output_args = shlex.split(add_to_output_cmd)
        subprocess.run(add_to_output_args)
```

Reply:
It could work, but it will update $GITHUB_OUTPUT in docker image
Will Github Action workflow retrieve the variable from inside the docker image ? if not we have to find another way (maybe some dummy file in artifacts that we could check for existence ...

Reply:
Yes, the existing code looks correct, just deprecated. The goal here is simply to append to the file named `GITHUB_OUTPUT`, so we can achieve that without spawning a process.

```py
from os import environ

output_file = environ["GITHUB_OUTPUT"]
with open(output_file, "a", encoding="utf-8") as output_stream:
    print(f"has_updated_sources={str(self.has_updated_sources)}", file=output_stream)
```

Reply:
@Kurt-von-Laven print(), not output_stream.write() ?

Reply:
The last line could be replaced with:

```py
output_stream.write(f"has_updated_sources={str(self.has_updated_sources)}\n")
```

Any preference?

Just saw your previous message. To be clear, we don’t want to overwrite the environment variable `GITHUB_OUTPUT`, which is essentially the path to a magic dummy file that gets sourced at the beginning of each subsequent step.

Reply:
I'd prefer the write option , it's more readable that a print that we expect more to be just a line in the console :)

Reply:
@parkerbxyz maybe u'd like to ressuscitate your PR and update it ? :) ( ps: just been rickrolled... 😠 🤣 )

Reply:
Sounds good to me. I updated the code snippet accordingly.

Ha ha, the meme that will never die.